src/mewbot/core.py

```diff
@@ -151,8 +118,51 @@ async def process(self, event: InputEvent) -> None:
         pass
 
 
+Component = Union[
+    BehaviourInterface,
+    IOConfigInterface,
+    TriggerInterface,
+    ConditionInterface,
+    ActionInterface,
+]
+
+
+class ComponentKind(str, enum.Enum):
+    BEHAVIOUR = "Behaviour"
+    TRIGGER = "Trigger"
+    CONDITION = "Condition"
+    ACTION = "Action"
+    IO_CONFIG = "IOConfig"
+    TEMPLATE = "Template"
+    DATASOURCE = "DataSource"
+
+    @classmethod
+    def has_value(cls, value: str) -> bool:
+        return value in cls.values()
+
+    @classmethod
+    def values(cls) -> List[str]:
+        return list(e for e in cls)
+
+    @classmethod
+    def interface(cls, value: ComponentKind) -> Type[Component]:
+        _map: Dict[ComponentKind, Type[Component]] = {
+            cls.BEHAVIOUR: BehaviourInterface,
+            cls.TRIGGER: TriggerInterface,
+            cls.CONDITION: ConditionInterface,
+            cls.ACTION: ActionInterface,
+            cls.IO_CONFIG: IOConfigInterface,
+        }
+
+        if value in _map:
+            return _map[value]
+
+        raise ValueError(f"Invalid value {value}")
+
+
 __all__ = [
```

Comment:
Should these have a particular ordering, i.e. same as in file for findability?
 I cannot fathom the suggestions interface so you'll have to give me a bit for a suggestion.

Reply:
I'm not sure it matters, really? The order in the file is to prevent forward definition issues, the order here was mostly grouped by purpose.
There's no guarantee that the things being exported are even first defined in this file (part of this PR is in fact exporting things from another module)

Reply:
I think, in the general case, re-ordering to increase clarity is a good thing. Not sure how to do it here though. But we should be mindful of that in future - I've seen people draw deep - largely wrong - inferences from structure.

Reply:
I think it could just be left as developer preference and is far and away not a thing we should worry about. I'd put it in the level of minor readability.

Reply:
Makes sense! We can always change it later on a readability pass.